### DISCUSSION TITLE
Retry connection

### DISCUSSION THREAD

**Discussion by Spo-oky** [obs-websocket-community-projects/obs-websocket-js#284] [Q&A]
Hi,

I'm starting to code and trying to connect to OBS to retrieve data from an OBS but this is done in automatic.

But the machine can be shutdown or OBS disconnected, is there a way to make a reconnection from time to time to connect as soon as OBS is open?

Thanks

**Comment by t2t2** [comment]
You can call `.connect` again with the same values after `ConnectionClosed` event, which keeps all of the current event listeners. Add some logic to only do it after identified event, cancel after a few attempts and should be all good
